### DISCUSSION TITLE
Support For Vue 3?

### DISCUSSION THREAD

**Discussion by Chaiwat-R** [easepick/easepick#76] [Q&A] [answered]
Hi. i have two question
-easepick support for vue.js 3 ?
- For now i use Litepicker. Is that possible? I wanna change text in year dropdown to Buddhist Era. but value is still Christian Era.
#Example I selcet this year in dropdown it will show 2565 but value is still 2022 ( in select option value is 2022 but Text is 2565 )
because In Thailand use Buddhist Era.
If is that possible how to make it. I use this in Vue.js 3 i wanna know how to use it?
Use in Litepicker version 2.0.11

**Comment by wakirin** [comment, marked as answer]
> easepick support for vue.js 3 ?

Should work. If you are using nuxt.js then remember to use client-only.

> For now i use Litepicker. Is that possible? I wanna change text in year dropdown to Buddhist Era.

you need to update html elements after view. eg.:
```js
...
setup(picker) {
  picker.on('view', (evt) => {
    const { view, target } = evt.detail;

    if (view === 'CalendarHeader') {
      // here find element and update
      // const element = target.querySelector(...)
    }
  });
}
...
```

**Comment by Chaiwat-R** [comment]
Great to hear that. but I don't use Nuxt.js .
In second question i use **### Litepicker**. it not  have event **view** 
![image](https://user-images.githubusercontent.com/106660340/172829737-1f48e9df-b009-4b70-8499-31962fb7da2f.png)
how to update it or how to call option.text in this calendar.ts

**Comment by Floriansylvain** [comment]
Hello! In case you (and/or anyone else passing by) need help to use easepick in Vue3 (with Composition API), here is a little tip.

If you want to use a plugin like range-plugin, you'll have to install with npm `@easepick/core` and `@easepick/range-plugin`, then import them in your component, for example:
```ts
import { easepick } from "@easepick/core"
import { RangePlugin } from "@easepick/range-plugin"
```
Then, to create your calendar:

In your template
```html
<input id="checkin"/>
<input id="checkout"/>
```

In your ``<script setup>``
```ts
onMounted(() => {
    const picker = new easepick.create({
        element: "#checkin",
        css: [
            "https://cdn.jsdelivr.net/npm/@easepick/bundle@1.2.0/dist/index.css"
        ],
        zIndex: 20,
        lang: "fr-FR",
        grid: 2,
        calendars: 2,
        RangePlugin: {
            elementEnd: "#checkout",
            delimiter: " / "
        },
        plugins: [
            RangePlugin
        ]
    })
})
```

If anyone has a better way to do this, I'd love to hear it!

**Reply by itwondersteam** [reply]
@Floriansylvain are you able to listen to on change event via `@change`  callback ? tried below but onChange doesn't get fired

```html
 <input :id="calendarRef" @change="(event:Event) => onChange((event.target as HTMLInputElement).value)">
```

**Reply by Floriansylvain** [reply]
I think @kenng found the solution https://github.com/easepick/easepick/discussions/76#discussioncomment-5727924

**Comment by kenng** [comment]
This is currently how we are using it with vue

```js
const emit = defineEmits(['change'])

const calendarRef = computed(() => 'easePick-' + props.id)

onMounted(() => {
    const picker = new easepick.create({
        element: '#' + calendarRef.value,
        css: ['https://cdn.jsdelivr.net/npm/@easepick/core@1.2.1/dist/index.css'],
        zIndex: 10,
        setup(picker) {
            picker.on('select', (e) => {
                emit('change', e.detail.date)
            });
        },
    })
})
```
